### PR TITLE
Added a role to hazelcastConfiguration so that hazelcast can be used …

### PR DESCRIPTION
**Pull request by makosblade**
…as a client or a server node. Also made some methods static and cleaned a little formatting

### CODE REVIEW

**Comment by geekbeast** [comment]
LGTM

File: src/main/java/com/geekbeast/rhizome/configuration/hazelcast/HazelcastConfiguration.java [hunk 1]
```diff
@@ -33,6 +35,7 @@
     private final String              group;
     private final String              password;
     private final int                 port;
+    private final String              role;
```
Comment (geekbeast) [inline src/main/java/com/geekbeast/rhizome/configuration/hazelcast/HazelcastConfiguration.java]: I feel like this makes more sense as a boolean as there only two possible roles.

File: src/main/java/com/geekbeast/rhizome/pods/hazelcast/BaseHazelcastInstanceConfigurationPod.java [hunk 2]
```diff
@@ -46,25 +48,46 @@ public Config getHazelcastConfiguration() {
         return config;
     }
 
-    protected NetworkConfig getNetworkConfig( HazelcastConfiguration hzConfiguration ) {
+    @Bean
```
Comment (geekbeast) [inline src/main/java/com/geekbeast/rhizome/pods/hazelcast/BaseHazelcastInstanceConfigurationPod.java]: Shouldn't be exposed as a bean as changing after hazelcast is created will have no effect.

File: src/main/java/com/geekbeast/rhizome/pods/hazelcast/RegistryBasedHazelcastInstanceConfigurationPod.java [hunk 3]
```diff
@@ -60,6 +62,27 @@ public Config getHazelcastConfiguration() {
         return config;
     }
 
+    @Bean
```
Comment (geekbeast) [inline src/main/java/com/geekbeast/rhizome/pods/hazelcast/RegistryBasedHazelcastInstanceConfigurationPod.java]: Same as above, shouldn't be a bean.

File: src/main/java/com/geekbeast/rhizome/pods/hazelcast/BaseHazelcastInstanceConfigurationPod.java (hunk 2 above)
Comment (makosblade) [inline src/main/java/com/geekbeast/rhizome/pods/hazelcast/BaseHazelcastInstanceConfigurationPod.java]: I'm not sure I understand why this shouldn't be exposed this way. Don't I need to do this to be able to @Inject it in the HazelcastPod ?

Comment (geekbeast) [inline src/main/java/com/geekbeast/rhizome/pods/hazelcast/BaseHazelcastInstanceConfigurationPod.java]: `@Inject` declares a dependency on `ClientConfig` for the `HazelcastPod`. This means that a client configuration must be present for the server to start, even if its intended to be run in server mode. It sounds like we need a class with two `Optional<...>` parameters for configuration. That class should be the bean exposed to the HazelcastPod. It should fail if there is not at least one of the necessary configuration classes provided.

Comment (makosblade) [inline src/main/java/com/geekbeast/rhizome/pods/hazelcast/BaseHazelcastInstanceConfigurationPod.java]: Gotcha, makes sense.

**Pull request merged**